core/modules/replica/FileUtils.cc

```diff
@@ -279,6 +279,44 @@ string FileUtils::createTemporaryFile(string const& baseDir,
 }
 
 
+void FileUtils::verifyFolders(string const& requestorContext,
+                              vector<string> const& folders,
+                              bool createMissingFolders) {
+    string const context = "FileUtils::" + string(__func__) + "(" + requestorContext + ") ";
+    for (auto&& folder: folders) {
+        if (folder.empty()) {
+            throw invalid_argument(context + " the empty folder name found in the collection.");
+        }
+        fs::path const path(folder);
+        if (!path.is_absolute()) {
+            throw invalid_argument(context + " non-absolute path '" + folder + "' found in the collection.");
+        }
+        boost::system::error_code ec;
+        if (createMissingFolders) {
+            if (!fs::create_directories(path, ec)) {
+                if (ec.value() != 0) {
+                    throw runtime_error(
+                            context + " failed to create folder '" + folder
+                            + "' or its intermediate subfolders, error: " + ec.message());
+                }
+
+            }
+        }
+
+        // Create a temporary file to test write permissions set for the folder.
+        // The file will get removed upon the completion of the test.
+        try {
+            string const tmpFileName = createTemporaryFile(folder, ".test-write-permissions-");
+            fs::remove(fs::path(tmpFileName), ec);
```

Comment:
This will not throw if you pass `ec`? The message below implies that you also expecting exception from `remove`?

Reply:
Actually, I don't care about any exceptions thrown by `remove as there is nothing that could be done here. Leaving undeleted files in a folder is not going to be a problem since there is no such file created in the relevant folders when the system is starting. The only possible, yet highly unlikely, scenario (apart from the underlying file system going down at a time of the file removal request or some other system error) would be another process deleting the same file. This may happen if two instances of the Replication ingest system were running on the same set of folders AND if two processes managed to generate exactly the same file name using the temporary file generator `createTemporaryFile`.
The actual reason for the `try`  block here is the previous line - method `createTemporaryFile` that is making some the temporary file.

Reply:
this is fine with me, the exception message below says `" failed to create/delete the temporary file ..."` which I interpreted as "create OR delete". Maybe just don't mention "delete" there?

Reply:
👍